### PR TITLE
YETUS-1030. Support linecomments in junit

### PR DESCRIPTION
**Pull request by aw-was-here**

### CODE REVIEW

File: asf-site-src/source/documentation/in-progress/precommit/plugins/junit-bugsystem.html.md [hunk 1]
```diff
@@ -41,8 +41,16 @@ None
...
 |:---------|:------|
+| `--junit-report-style=[full|line]` | Style of the junit report |
 | `--junit-report-xml=<file>` | Name of the output file |
 
+## JUnit Style
+
+The JUnit report type has two formats:
+
+* `full` - summarizes per-test and provides a link to the report for that test
+* `line` - summarizes per-file and works better with integrated CI systems and with external readers such as Jenkins WarningNG plug-in
```
Comment (ndimiduk) [inline asf-site-src/source/documentation/in-progress/precommit/plugins/junit-bugsystem.html.md:52]: What is an "integrated CI system" ?

File: precommit/src/main/shell/plugins.d/junit.sh [hunk 2]
```diff
@@ -25,16 +25,21 @@ JUNIT_FAILED_TESTS=""
 JUNIT_TEST_OUTPUT_DIR="."
 JUNIT_TEST_PREFIX="org.apache."
 
+JUNIT_REPORT_STYLE="line"
```
Comment (ndimiduk) [inline precommit/src/main/shell/plugins.d/junit.sh:28]: Are we sure we want change the default report style on people? Needs a big release note...

**Review by ndimiduk — APPROVED** [review]
Tentative +1 ... I'm not sure about changing the default, why line-oriented reports are better.

File: asf-site-src/source/documentation/in-progress/precommit/plugins/junit-bugsystem.html.md (hunk 1 above)
Comment (aw-was-here) [reply asf-site-src/source/documentation/in-progress/precommit/plugins/junit-bugsystem.html.md:52]: Semaphore, Gitlab, Jenkins, and more have the ability to reference the source code if the proper junit incantations are used.

File: precommit/src/main/shell/plugins.d/junit.sh (hunk 2 above)
Comment (aw-was-here) [reply precommit/src/main/shell/plugins.d/junit.sh:28]: Looking at a lot of different junit output implementations over the past week, our current form is very much outside the norm. There will definitely be a release note, but long term I feel strongly that this approach is generally better.

Comment (aw-was-here) [reply precommit/src/main/shell/plugins.d/junit.sh:28]: (One thing to keep in mind is that our current approach basically hides how many _actual_ errors there are; instead it is a count of how many tests failed!)

**Comment by aw-was-here** [comment]
Thanks for the review.  I totally get the tentative +1.  junit, as a format, is a mess. :frowning:

**Pull request merged**